src/subscription/__tests__/mapAsyncIterator-test.js

```diff
@@ -0,0 +1,193 @@
+    async function* source() {
+      yield 1;
+      yield 2;
+      yield 3;
+    }
+
+    const doubles = mapAsyncIterator(source(), x => x + x);
+
+    expect(
+      await doubles.next()
+    ).to.deep.equal({ value: 2, done: false });
+    expect(
+      await doubles.next()
+    ).to.deep.equal({ value: 4, done: false });
+
+    // Early return
+    expect(
+      await doubles.return()
+    ).to.deep.equal({ value: undefined, done: true });
+
+    // Subsequent nexts
+    expect(
+      await doubles.next()
+    ).to.deep.equal({ value: undefined, done: true });
+    expect(
+      await doubles.next()
+    ).to.deep.equal({ value: undefined, done: true });
+  });
+
+  it('passes through early return from async values', async () => {
+    async function* source() {
+      try {
+        yield 1;
+        yield 2;
+        yield 3;
+      } finally {
+        yield 'done';
+        yield 'last';
+      }
+    }
+
+    const doubles = mapAsyncIterator(source(), x => x + x);
+
+    expect(
+      await doubles.next()
+    ).to.deep.equal({ value: 2, done: false });
+    expect(
+      await doubles.next()
+    ).to.deep.equal({ value: 4, done: false });
+
+    // Early return
+    expect(
+      await doubles.return()
+    ).to.deep.equal({ value: 'donedone', done: false });
+
+    // Subsequent nexts may yield from finally block
+    expect(
+      await doubles.next()
+    ).to.deep.equal({ value: 'lastlast', done: false });
+    expect(
+      await doubles.next()
+    ).to.deep.equal({ value: undefined, done: true });
+  });
+
+  it('allows throwing errors through async generators', async () => {
+    async function* source() {
+      yield 1;
+      yield 2;
+      yield 3;
+    }
+
+    const doubles = mapAsyncIterator(source(), x => x + x);
+
+    expect(
+      await doubles.next()
+    ).to.deep.equal({ value: 2, done: false });
+    expect(
+      await doubles.next()
+    ).to.deep.equal({ value: 4, done: false });
+
+    // Throw error
+    let caughtError;
+    try {
+      await doubles.throw('ouch')
+    } catch (e) {
+      caughtError = e;
+    }
+    expect(caughtError).to.equal('ouch');
+
+    expect(
+      await doubles.next()
+    ).to.deep.equal({ value: undefined, done: true });
+    expect(
+      await doubles.next()
+    ).to.deep.equal({ value: undefined, done: true });
+  });
+
+  it('passes through caught errors through async generators', async () => {
+    async function* source() {
+      try {
+        yield 1;
+        yield 2;
+        yield 3;
+      } catch (e) {
+        yield e;
+      }
+    }
+
+    const doubles = mapAsyncIterator(source(), x => x + x);
+
+    expect(
+      await doubles.next()
+    ).to.deep.equal({ value: 2, done: false });
+    expect(
+      await doubles.next()
+    ).to.deep.equal({ value: 4, done: false });
+
+    // Throw error
+    expect(
+      await doubles.throw('ouch')
+    ).to.deep.equal({ value: 'ouchouch', done: false });
```

Comment:
This test needs more comments. The map function doesn't apply to error handling - it only applies to yielded values. This test is checking an edge case where if the parent generator catches an error and continues that the continued yielded values are also mapped